code/modules/projectiles/guns/flamer.dm

```diff
@@ -313,6 +313,31 @@
 /obj/item/weapon/gun/flamer/som/mag_harness
 	starting_attachment_types = list(/obj/item/attachable/flamer_nozzle/wide, /obj/item/attachable/magnetic_harness)
 
+//dedicated engineer pyro kit flamer
+/obj/item/weapon/gun/flamer/big_flamer/engineer
+	name = "\improper FL-242 incinerator unit"
+	desc = "The FL-242 is a more light weight incinerator unit designed specifically to fit into its accompanying engineers bag. Can only be used with back mounted fuel tanks however."
+	default_ammo_type = null
+	allowed_ammo_types = list(
+		/obj/item/ammo_magazine/flamer_tank/backtank,
+		/obj/item/ammo_magazine/flamer_tank/backtank/X,
+		/obj/item/ammo_magazine/flamer_tank/internal,
+	)
+
+/obj/item/weapon/gun/flamer/big_flamer/engineer/MouseDrop_T(obj/item/W, mob/living/user) //Dragging the backpack to the gun will load the gun with the internal mag.
+	. = ..()
+	if(!istype(W, /obj/item/storage/holster/backholster/flamer))
+		return
+
+	var/obj/item/storage/holster/backholster/flamer/D = W
+	var/obj/item/ammo_magazine/flamer_tank/internal/tank = D.tank
+	tank.forceMove(src.loc)
```

Comment:
I don't know what's going on here but I don't like it.

Reply:
For the flamers ammo supply to work it needs to be on the persons loc, otherwise I need to alter a really important function and thats bad